common/src/main/java/com/hedera/block/common/utils/FileUtilities.java

```diff
@@ -166,17 +171,38 @@ public static byte[] readGzipFileUnsafe(@NonNull final Path filePath) throws IOE
      * <p>
      * This method uses default extensions for gzip and block files.
      * <p>
-     * This method is _unsafe_ because it reads the entire file content into
-     * a single byte array, which can cause memory issues, and may fail if the
-     * file contains a large amount of data.
+     * This method is _unsafe_ because it reads the entire file content into a single byte array,
+     * which can cause memory issues, and may fail if the file contains a large amount of data.
+     *
+     * @param file to read bytes from
+     *
+     * @return byte array of the content of the file or null if the file extension is not supported
+     *
+     * @throws IOException      if unable to read the file.
+     * @throws OutOfMemoryError if a byte array large enough to contain the file contents cannot be
+     *                          allocated (either because it exceeds MAX_INT bytes or exceeds
+     *                          available heap memory).
+     */
+    public static byte[] readFileBytesUnsafe(@NonNull final File file) throws IOException {
+        return readFileBytesUnsafe(file.toPath());
+    }
```

Comment:
I do not believe this is a good change, generally you had to use a Path to get the File in the first place, and it's better to pass in the Path rather than converting the Path to a File and back to a Path..

Reply:
I did this for convenience, since the only place I use that, I need to call `blockFile.getPath()`, and thought that it would be good to simply have this convenience where I do not need to call `getPath()` on a `File` object so that I can read the bytes. If you do really think this is not a good idea at all, I will remove this, but I do believe even in the future we will have places where the local scope only has the `File` object visible and we will need to call `File#getPath()` somewhere to get it.

Reply:
`File` is an older API that should be replaced by `Path` in nearly all cases.
In most cases the construction of a File is either in unit tests, or a leftover from older API assumptions.

Note, the one place (in non-test code) I see you using this, the File is constructed from Path and filename.  A Path should be `resolve`d _instead_ and passed in, avoiding _any_ construction of the `File` object (and making the caller more efficient in general).

Example:
We have (in relevant part)
```java
        final File blockFile = new File(blockstreamPath, nextBlockFileName);

        if (!blockFile.exists()) {
            return null;
        }

        try {
            final byte[] blockBytes = FileUtilities.readFileBytesUnsafe(blockFile);
```
Should be
```java
        final Path blockFile = blockstreamPath.resolve(nextBlockFileName);
        if (!Files.isReadable(blockFile)) {
            return null;
        }
        try {
            final byte[] blockBytes = FileUtilities.readFileBytesUnsafe(blockFile);
```

Reply:
I updated my example as you have proposed, have left a todo also in case someone wants to comment on the possibility of a null value now returned by our `common` lib method usage, let me know also your opinion, the todo is [here](https://github.com/hashgraph/hedera-block-node/pull/271/files#r1818628609).